### DISCUSSION TITLE
Please let tycho-p2-extras-plugin support mirroring a repository using its mirrors

### DISCUSSION THREAD

**Discussion by Baytars** [eclipse-tycho/tycho#1431] [Ideas]
I need to mirror a repository (A, https://download.eclipse.org/xxx/yyy). However, I don't want to access that repository because the transfer speed is slow. So I turned to `<mirror>` tag in `pom.xml`, only to find that the mirror (B, https://mirrors.aliyun.com/eclipse/xxx/yyy) only takes effect for those repositories written under `<repositories>` in `pom.xml`.
Mirroring a repository (A, https://download.eclipse.org/xxx/yyy) using its mirror (B, https://mirrors.aliyun.com/eclipse/xxx/yyy), sounds like a brain-twister, right? But please imagine a scenario: we are going to mirror B directly using `tycho-p2-extras-plugin`, instead. However, mirror B is a composite repository that refers to other repositories under the same domain name as A (C, https://download.eclipse.org/zzz/kkk). We would also like to mirror C using its mirror (D, https://mirrors.aliyun.com/eclipse/zzz/kkk). If the `tycho-p2-extras-plugin` supports mirroring, life will can be a lot easier. Do you agree?

**Comment by laeubi** [comment]
I must confess I do not fully understand what should be archived, would it be possible to came up with an example project instead of all these A, B, C, Ds?
